### PR TITLE
fix: operator cannot install keycloak when resourcequota scope is not set (#492)

### LINKED ISSUE #492: operator fails to install keycloak when resourcequota scope is terminating or no scope is set.
**Describe the bug**
when creating ArgoCD instances in user workload projects and defining `.spec.sso.provider: keycloak` the operator is creating a `DeploymentConfig` which is leading to a stuck deploy Pod due to missing resources in the `DepolymentConfig` and Quotas enforcing requests and limits set.
Please make sure, that the `DeploymentConfig` has resources set in the strategy or please migrate to a `StatefulSet` like the RH-SSO is creating the keycloak to make it possible to run keycloak in environments where quotas are enforces also for NotTerminating Pods.

**To Reproduce**
1. Create a ns with `resourcequota`.
2. Do not set any scope to the `resourcequota`.
3. Install or run the operator locally using `make run`.
4. Install keycloak by updating the Argo CD spec -> `.sso.provider: keycloak`.
5. You can see that keycloak deployment fails as the `keycloak-deploy` pod does not have any resource requests/limits set.

**Expected behavior**
operator should install and configure keycloak for any `resourcequota` scope.

**Screenshots**
NA

**Additional context**
NA

### PR DESCRIPTION
**Pull request by iam-veeramalla**
Signed-off-by: iam-veeramalla <abhishek.veeramalla@gmail.com>

**What type of PR is this?**
> /kind bug

**What does this PR do / why we need it**:
Fixes #492 

**Have you updated the necessary documentation?**
NA

**Which issue(s) this PR fixes**:
Fixes #492 

**How to test changes / Special notes to the reviewer**:
Please refer to the issue #492

### CODE REVIEW

**Comment by iam-veeramalla** [comment]
We dont need any reconciliation logic or a place holder in the CR as the lifetime of the deployer pod is only till the keycloak service is up and running. 

Background: On an OCP cluster, We use DeploymentConfig to install keycloak. DeploymentConfig creates a special pod called as a deployer pod. The scope or a lifetime of a deployer pod is only till the actual keycloak pod(container) is up and running. Once the keycloak pod is running, the deployer pod moves in to the `completed` state. So it is a terminating pod(pod with a limited lifetime).

Usually in most of the cases, `resourcequota` resource is created with a scope set to `NotTerminating`, which means the resource requirements is only made mandatory to the pods that have unlimited lifetime, so `deployer pods` do not fall into this category. However, In some cases, customers have `resourcequota` set to scope `Terminating`, probably because their pods have a lifetime associated. 

To overcome this issue, This PR sets a minimum resource requirements to the deployer pod. I don't see any reason to modify the resource requirements of deployer pod. so, I did not provide any option in the CR to override the default values.

**Comment by iam-veeramalla** [comment]
PTAL @jannfis

File: controllers/argocd/keycloak.go [hunk 1]
```diff
@@ -305,6 +305,16 @@ func getKeycloakDeploymentConfigTemplate(cr *argoprojv1a1.ArgoCD) *appsv1.Deploy
...
 				Type: "Recreate",
+				Resources: corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						corev1.ResourceMemory: resourcev1.MustParse("256Mi"),
+						corev1.ResourceCPU:    resourcev1.MustParse("250m"),
+					},
+					Requests: corev1.ResourceList{
+						corev1.ResourceMemory: resourcev1.MustParse("512Mi"),
+						corev1.ResourceCPU:    resourcev1.MustParse("500m"),
+					},
```
Comment (jannfis) [inline controllers/argocd/keycloak.go:316]: I think having requests higher than limits make no sense - they should be swapped.

Comment (iam-veeramalla) [reply controllers/argocd/keycloak.go]: My bad, made the required changes. PTAL @jannfis

**Review by wtam2018 — APPROVED** [review]
LGTM

**Pull request merged**